code/scripting/api/objs/hudconfig.cpp

```diff
@@ -52,6 +52,86 @@ bool hud_preset_h::isValid() const
 	return preset >= 0 && preset < (int)HC_preset_filenames.size();
 }
 
+hud_color_preset_h::hud_color_preset_h() : preset(-1) {}
+hud_color_preset_h::hud_color_preset_h(int l_preset) : preset(l_preset) {}
+
+int hud_color_preset_h::getIndex() const
+{
+	return preset;
+}
+
+SCP_string hud_color_preset_h::getName() const
+{
+	if (!isValid()) {
+		return "";
+	}
+
+	return XSTR(HC_colors[preset].name.c_str(), HC_colors[preset].xstr);
+}
+
+bool hud_color_preset_h::isValid() const
+{
+	return preset >= 0 && preset < (int)HC_preset_filenames.size();
+}
+
+//**********HANDLE: hud color preset
+ADE_OBJ(l_HUD_Color_Preset, hud_color_preset_h, "hud_color_preset", "Hud preset handle");
+
+ADE_VIRTVAR(Name, l_HUD_Color_Preset, nullptr, "The name of this preset", "string", "The name")
+{
+	hud_color_preset_h current;
+	if (!ade_get_args(L, "o", l_HUD_Color_Preset.Get(&current))) {
+		return ADE_RETURN_NIL;
+	}
+
+	if (!current.isValid()) {
+		return ade_set_error(L, "s", "");
+	}
+
+	if (ADE_SETTING_VAR) {
+		LuaError(L, "This property is read only.");
+	}
+
```

Comment:
Not sure how this question didn't get saved.  Is there any value in placing this above the is valid check?  It might save someone some time in the future to know that they shouldn't be using the function this way, even before they find out that the object is invalid.

Reply:
I dunno. As one who writes a lot of scripts, I think it's more important to know that I'm working with in invalid object before I know if it can be modified. The former is a more critical bug in my Lua code, if that makes sense.

Reply:
I was trying to figure out what other changes you were referring to... turns out I have to actually push the updated code. 😅